src/thor/alternates.cc

```diff
@@ -0,0 +1,128 @@
+#include <iostream>
+#include <vector>
+
+#include "thor/alternates.h"
+
+using namespace valhalla::thor;
+using namespace valhalla::baldr;
+using namespace valhalla::midgard;
+
+/*
+ * Viability tests for alternate paths based on M. Kobitzsch's Alternative Route
+ * Techniques (2015). Tests verify limited sharing between segments, bounded
+ * stretch, and local optimiality. Any candidate path that meets all the criteria
+ * may be considered a valid alternate to the shortest path.
+ */
+namespace {
+// Defaults thresholds
+float kAtMostLonger = 1.25f; // stretch threshold
+float kAtMostShared = 0.75f; // sharing threshold
+// float kAtLeastOptimal = 0.2f; // local optimality threshold
```

Comment:
we didnt implement local optimality and the paper actual mentions that sometimes it removes some good results. at the end of the day an implementation that supports that was too painful to stomache at the moment.